File: Front/TS/src/infrastructure/ui/Angular/src/presenters/AddFriendPresenter.ts
import { FriendsController } from "../../../../adapters/controllers/FriendsController";
import type { IAddFriendOutputPort } from "../../../../../domain/features/add-friend/AddFriendHandler";
import type { MemberDto } from "../../../../../domain/members/IMemberRepository";

export class AddFriendPresenter implements IAddFriendOutputPort {
    private controller: FriendsController|undefined;
    Present(memberId: string): void {
        const member = (this.controller!.vm.Members as MemberDto[]).find(x => x .Id == memberId)
        if(member) {
            member.IsFriend = true;
            this.controller?.vm.Friends.push({
                Id: member.Id,
                Name: member.Name
            });
        }
    }

    Subscribe(controller: FriendsController) {
        this.controller = controller;
    }
}

export interface FriendAdded {
    MemberId: string;
}

File: Front/.Net Maui/BetFriends/Features/Friends/AddFriendPresenter.cs
using BetFriend.Domain.Features.AddFriend;
using CommunityToolkit.Mvvm.Messaging;

namespace BetFriends.Features.Friends;

internal class AddFriendPresenter : IAddFriendOutputPort
{
    public void Present(string id)
    {
        WeakReferenceMessenger.Default.Send(new FriendAdded(id));
    }
}

internal record FriendAdded(string Id);